### ISSUE TITLE
all about 1e-7

### ISSUE THREAD

**Issue by 527760681** [kkweon/UNet-in-Tensorflow#4]
hi,i downloaded your project for study,and thank you for this super useful code.
i have two problem as follows:
1.the "relative path" code not work on my windows machine,such as data.py,i have to change it to absolute path,that's sad.
2.what does the 1e-7 means?i knew it means 0.0000001,but why you always add 1e-7 behind your code ?
i'm new to AI and Tensorflow,my english is not very good,sorry about that.
hope you reply soon.
thanks again.

**Comment by kkweon** [comment]
Hi thanks for the issue

1. Did you run from the project root directory?
Or did you run inside the util directory? This wont work because of the relative import.

you should always use from the project root such as 

from utils import ,,,

2. It is for numerical stability. Because pure 0 can cause some NaN for example dividing by zero.
So 1e-7 is a little safety feature to prevent pure zero. 

Hope that helps!

**Comment by 527760681** [comment]
thank you for reply.
after train on my  GTX1080 from 1:30 a.m. to 4.30 a.m., i found that the result of prediction is not very good in seeing the result picture.how's your result? 
my tensorboard shows the IOU as follow:

![qq 20180122131131](https://user-images.githubusercontent.com/16307769/35206277-87bff8f6-ff76-11e7-99f3-b9718babe99b.png)

not same as classifacation,this IOU performs good or not?
and what should i do if i want to change the net back to the real UNET?just change the loss func into pixel-wise softmax? 
wish you have a nice day~
hope for reply.

**Comment by kkweon** [comment]
Yes, I had to make some change lately because the datasets (labels) has changed. Since then, I am also having the same issue. Anyway, I was also going back to the pixel-wise softmax, so yeah you need to change the loss function and it's good :smile:

**Comment by blackhawkmc** [comment]
Hello @kkweon and thank you for the code, can you please provide us, with new code you used or changes you made for softmax? Please!
